metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -189,202 +152,237 @@ class ScalametaLanguageServer(
       renameProvider = true,
       codeActionProvider = true
     )
-    InitializeResult(capabilities)
+    Task(Right(InitializeResult(capabilities)))
   }
 
-  override def shutdown(): Task[ShutdownResult] = Task {
+  // TODO(olafur): make it easier to invoke fluid services from tests
+  def shutdown(): Unit = {
     logger.info("Shutting down...")
     cancelEffects.foreach(_.cancel())
-    connection.cancelAllActiveRequests()
-    ShutdownResult()
   }
 
-  private def onChangedFile(
-      path: AbsolutePath
-  )(fallback: AbsolutePath => Unit): Unit = {
-    val name = PathIO.extension(path.toNIO)
-    logger.info(s"File $path changed, extension=$name")
-    name match {
-      case "semanticdb" => fileSystemSemanticdbSubscriber.onNext(path)
-      case "compilerconfig" => compilerConfigSubscriber.onNext(path)
-      case _ => fallback(path)
+  val services: Services = Services.empty
+    .requestAsync[InitializeParams, InitializeResult]("initialize") { params =>
+      initialize(params)
     }
-  }
-
-  override def onChangeWatchedFiles(changes: Seq[FileEvent]): Unit =
-    changes.foreach {
-      case FileEvent(
-          Uri(path),
-          FileChangeType.Created | FileChangeType.Changed
-          ) =>
-        onChangedFile(path.toAbsolutePath) { _ =>
-          logger.warn(s"Unknown file extension for path $path")
+    .request[JsValue, JsValue]("shutdown") { _ =>
```

Comment:
I think this is wrong, to some degree: some requests are expected to return `null` in the success case, like this one (`shutdown`), and I would use `JsNull.type` explicitly to make sure we are not returning a value. In this case `shutdown` also takes no parameters, so `JsNull.type` is probably appropriate for the request params too.

In other cases `JsValue` is appropriate because the request params is of type `any`, like the `DidChangeConfiguration` notification.

Reply:
In circe there is not JsNull.type. I tried JsNull.type but the implicit resolution failed (no idea why) so I used JsValue instead.

Note that empty params fields are converted to a `Json.Null` so there is no way to distinguish empty params vs. `"params" : null`